src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_LateBindingHelpers.vb

```diff
@@ -990,16 +986,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' </summary>
         Private Function TryGetSpecialMember(Of T As Symbol)(<Out> ByRef result As T,
                                                        memberId As SpecialMember,
-                                                       syntax As SyntaxNode) As Boolean
+                                                       syntax As SyntaxNode,
+                                                       Optional isOptional As Boolean = False) As Boolean
 
             result = Nothing
             Dim useSiteInfo As UseSiteInfo(Of AssemblySymbol) = Nothing
             Dim memberSymbol = Binder.GetSpecialTypeMember(Me._topMethod.ContainingAssembly, memberId, useSiteInfo)
 
-            If Binder.ReportUseSite(_diagnostics, syntax.GetLocation(), useSiteInfo) Then
+            If useSiteInfo.DiagnosticInfo IsNot Nothing Then
+                If Not isOptional Then
```

Comment:
This is closer to the behavior I'd expect to see in C#; either we report the warning for the special member, or we just bail in the presence of any use-site diagnostic, regardless of whether it's an error.

Reply:
According to Aleksey, there are no use site warnings in VB. In such case "we just bail in the presence of any use-site diagnostic, regardless of whether it's an error" should be equvivalent to C#'s "bail on errors and ignore warnings" logic

Reply:
This is the pattern from the function right above this one, ```TryGetWellknownMember```.